Cann't build dev-lang/rust-9999 with tools enabled.

After merging https://github.com/rust-lang/rust/pull/40584 rust became unavailable to build with extended tools:

```
executing step Step { name: "tool-rls", stage: 2, host: "x86_64-unknown-linux-gnu", target: "x86_64-unknown-linux-gnu" }
Building stage2 tool rls (x86_64-unknown-linux-gnu)
running: "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "build" "-j" "8" "--target" "x86_64-unknown-linux-gnu" "-v" "--release" "--manifest-path" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/rls/Cargo.toml"

error[E0463]: can't find crate for `std`

error: aborting due to previous error

error: Could not compile `utf8-ranges`.

Caused by:
  process didn't exit successfully: `/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/bootstrap/debug/rustc --crate-name utf8_ranges /tmp/portage/dev-lang/rust-9999/homedir/.cargo/registry/src/github.com-1ecc6299db9ec823/utf8-ranges-1.0.0/src/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 -C metadata=ac45e277014bbc8a -C extra-filename=-ac45e277014bbc8a --out-dir /tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/deps --target x86_64-unknown-linux-gnu -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/deps -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-tools/release/deps --cap-lints allow` (exit code: 101)
Build failed, waiting for other jobs to finish...
error[E0463]: can't find crate for `std`

error: aborting due to previous error

error[E0463]: can't find crate for `std`

error: aborting due to previous error

error: Could not compile `diff`.

Caused by:
  process didn't exit successfully: `/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/bootstrap/debug/rustc --crate-name diff /tmp/portage/dev-lang/rust-9999/homedir/.cargo/registry/src/github.com-1ecc6299db9ec823/diff-0.1.10/src/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 -C metadata=448963b77c30997d -C extra-filename=-448963b77c30997d --out-dir /tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/deps --target x86_64-unknown-linux-gnu -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/deps -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-tools/release/deps --cap-lints allow` (exit code: 101)
Build failed, waiting for other jobs to finish...
error: Could not compile `matches`.

Caused by:
  process didn't exit successfully: `/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/bootstrap/debug/rustc --crate-name matches /tmp/portage/dev-lang/rust-9999/homedir/.cargo/registry/src/github.com-1ecc6299db9ec823/matches-0.1.4/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 -C metadata=3bb6e7f56ca2dd49 -C extra-filename=-3bb6e7f56ca2dd49 --out-dir /tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/deps --target x86_64-unknown-linux-gnu -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/deps -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-tools/release/deps --cap-lints allow` (exit code: 101)
Build failed, waiting for other jobs to finish...
error[E0463]: can't find crate for `std`

error: aborting due to previous error

error: Could not compile `semver`.

Caused by:
  process didn't exit successfully: `/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/bootstrap/debug/rustc --crate-name semver /tmp/portage/dev-lang/rust-9999/homedir/.cargo/registry/src/github.com-1ecc6299db9ec823/semver-0.1.20/src/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 -C metadata=d3411e58e18d556b -C extra-filename=-d3411e58e18d556b --out-dir /tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-tools/release/deps -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-tools/release/deps --cap-lints allow` (exit code: 101)
Build failed, waiting for other jobs to finish...
error[E0463]: can't find crate for `std`

error: aborting due to previous error

error[E0463]: can't find crate for `std`

error: aborting due to previous error

error: Could not compile `vec_map`.

Caused by:
  process didn't exit successfully: `/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/bootstrap/debug/rustc --crate-name vec_map /tmp/portage/dev-lang/rust-9999/homedir/.cargo/registry/src/github.com-1ecc6299db9ec823/vec_map-0.6.0/src/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 -C metadata=8b8258f4c07d94e0 -C extra-filename=-8b8258f4c07d94e0 --out-dir /tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/deps --target x86_64-unknown-linux-gnu -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/deps -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-tools/release/deps --cap-lints allow` (exit code: 101)
Build failed, waiting for other jobs to finish...
error: Could not compile `language-tags`.

Caused by:
  process didn't exit successfully: `/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/bootstrap/debug/rustc --crate-name language_tags /tmp/portage/dev-lang/rust-9999/homedir/.cargo/registry/src/github.com-1ecc6299db9ec823/language-tags-0.2.2/src/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 -C metadata=aa5a1721eca7257c -C extra-filename=-aa5a1721eca7257c --out-dir /tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/deps --target x86_64-unknown-linux-gnu -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/deps -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-tools/release/deps --cap-lints allow` (exit code: 101)
Build failed, waiting for other jobs to finish...
error[E0463]: can't find crate for `std`

error: aborting due to previous error

error[E0463]: can't find crate for `core`

error: aborting due to previous error

error: Could not compile `getopts`.

Caused by:
  process didn't exit successfully: `/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/bootstrap/debug/rustc --crate-name getopts /tmp/portage/dev-lang/rust-9999/homedir/.cargo/registry/src/github.com-1ecc6299db9ec823/getopts-0.2.14/src/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 -C metadata=76f4457fad24783c -C extra-filename=-76f4457fad24783c --out-dir /tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/deps --target x86_64-unknown-linux-gnu -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/deps -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-tools/release/deps --cap-lints allow` (exit code: 101)
Build failed, waiting for other jobs to finish...
error: build failed

command did not execute successfully: "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "build" "-j" "8" "--target" "x86_64-unknown-linux-gnu" "-v" "--release" "--manifest-path" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/rls/Cargo.toml"
expected success, got: exit code: 101

Build completed unsuccessfully in 0:22:12
make: *** [Makefile:66: dist] Error 1
```

--- a/dev-lang/rust/rust-9999.ebuild
+++ b/dev-lang/rust/rust-9999.ebuild
@@ -107,6 +107,15 @@ src_install() {
 	mv "${D}/usr/bin/rustdoc" "${D}/usr/bin/rustdoc-${PV}" || die
 	mv "${D}/usr/bin/rust-gdb" "${D}/usr/bin/rust-gdb-${PV}" || die
 	mv "${D}/usr/bin/rust-lldb" "${D}/usr/bin/rust-lldb-${PV}" || die
+	if use tools; then
+		mv "${D}/usr/bin/rls" "${D}/usr/bin/rls-${PV}" || die
+		# remove cargo
+		rm -f "${D}/usr/bin/cargo" || die
+		rm -f "${D}/usr/share/zsh/site-functions/_cargo" || die
+		rm -f "${D}/usr/share/rust-9999/man/man1/cargo*" || die
+		rm -f "${D}/etc/bash_completion.d/cargo" || die
+		rm -f "${D}/usr/lib64/rust-9999/rustlib/manifest-cargo" || die
+	fi
 
 	dodoc COPYRIGHT LICENSE-APACHE LICENSE-MIT
 

